driver/cmd/ibm-spectrum-scale-csi/main.go

```diff
@@ -143,14 +147,22 @@ func setContext() context.Context {
 	return ctx
 }
 
-func getLogEnv() string {
+func getLogEnv() (string, string) {
 	level := os.Getenv(logLevel)
-	return level
+	logLevel := strings.ToUpper(level)
+	if logLevel != TRACE.String() || logLevel != DEBUG.String() || logLevel != INFO.String() || logLevel != WARNING.String() || logLevel != ERROR.String() || logLevel != FATAL.String() {
+		logLevel = INFO.String()
+	}
+	persistentLogEnabled := os.Getenv(persistentLog)
+	if strings.ToUpper(persistentLogEnabled) != "ENABLED" {
+		persistentLogEnabled = "DISABLED"
+	}
+	return logLevel, persistentLogEnabled
 }
 
 func getLogLevel(level string) string {
 	var logValue string
-	if level == "" || level == DEBUG.String() || level == TRACE.String() {
+	if level == DEBUG.String() || level == TRACE.String() {
```

Comment:
The code around the setting of default log level as INFO is present in 2 functions getLogLevel and getLogEnv, ideally that should be handled by only one function and preferably by getLogLevel

Reply:
first condition is to check for user defined error like case sensitive and any other values than preferred values while getting env variable. Second condition is needed as there is no level called DEBUG and TRACE in klog which we are explicitly making to INFO to handle verbose feature.

Reply:
Also with getLogenv value we are printing trace "[a9518896-87a2-474d-baf9-7174a33ab31b] logValue: INFO" with level. If we modify in first place then always level would be seen as INFO. Already there is a defect for not showing FATAL,ERROR and WARNING with above log.

Reply:
Yeah that's why it makes more sense to use:
getLogEnv --> to get env var value as it and 
getLogLevel --> to handle the defaults in both the above cases, 
and we can add the print statement in getLogLevel saying something like "No/Incorrect value of logger is specifies, so defaulting to INFO", let's discuss if you have any other thoughts.. just trying to make sure the code to set defaults is in one place and so less error prone to further changes

Reply:
if we print that in getLogLevel then it would not write to file as we are setting flags and creating file later. I already tried this

Reply:
We discussed and moved defaults in getLogLevel and keeping the print in main